### PR TITLE
Refactor def_helper

### PR DESCRIPTION
**Pull request by tilk**
This PR untangles different use cases of `method_def_helper` and prepares the code for further re-use.

### CODE REVIEW

File: transactron/_utils.py [hunk 1]
```diff
@@ -122,24 +125,35 @@ def _graph_ccs(gr: ROGraph[T]) -> list[GraphCC[T]]:
 MethodLayout: TypeAlias = LayoutLike
 
 
-def method_def_helper(method, func: Callable[..., T], arg=None, /, **kwargs) -> T:
+def has_first_param(func: Callable[..., T], name: str, tp: type[U]) -> TypeGuard[Callable[Concatenate[U, ...], T]]:
+    parameters = signature(func).parameters
+    return (
+        len(parameters) > 1
+        and next(iter(parameters.keys())) == name
```
Comment (Arusekk) [inline transactron/_utils.py:132]: Would work even shorter:
```suggestion
        and next(iter(parameters)) == name
```
but might be less explicit.

**Review by Arusekk — APPROVED** [review]
I have one general comment. Otherwise LGTM

File: transactron/_utils.py [hunk 2]
```diff
@@ -122,24 +125,35 @@ def _graph_ccs(gr: ROGraph[T]) -> list[GraphCC[T]]:
...
     )
-    if (
-        len(parameters) == 1
-        and "arg" in parameters
-        and parameters["arg"].kind in {Parameter.POSITIONAL_OR_KEYWORD, Parameter.POSITIONAL_ONLY}
-        and parameters["arg"].annotation in {Parameter.empty, Record}
-    ):
-        if arg is None:
-            arg = kwargs
+    if len(parameters) == 1 and has_first_param(func, "arg", tp):
```
Comment (lekcyjna123) [inline transactron/_utils.py:143]: This condition will be always false. If `len(parameters)==1` will be true then `len(parameters) > 1` in `has_first_param` will be false, so `has_first_param` will return false.

Comment (tilk) [reply transactron/_utils.py:143]: Good catch. The condition in `has_first_param` should have been `>= 1`.

**Review by lekcyjna123 — APPROVED** [review]

**Pull request merged**